### ISSUE TITLE
problem about .\module-info.class and META-INF\services\9

### ISSUE THREAD

**Issue by PhoenixZeng** [moditect/moditect#240]
I am not good at English, so I use translation software 

From translation software:
My project makes use of a large number of unmodularized dependencies, so I used the maven-shade-plugin to package all of the dependencies into a single JAR file. Strangely, the maven-shade-plugin seems to delete the module-info.class file that I wrote myself. Therefore, I adopted this project to regenerate the module-info.class.

(Actually, regenerating it is the correct approach because excluding  dependencies in the module-info will cause compilation errors, and includingthem will result in runtime errors when running the JAR. Packaging everything together allows jlink to consider this project as having no external dependencies outside of jdk.)

During the generation process, I encountered an error stating that "this JAR is already modular." When I unzipped the JAR to inspect it, I found that module-info.class existed in META-INF\services\9, and this file originated from one of the dependency JARs. Ultimately, I had to exclude this dependency to make it work.

Although this issue may have arisen from the maven-shade-plugin, based on my understanding, setting overwriteExistingFiles to true should have allowed me to directly overwrite this file. Or rather, all modularization related files should be excluded and only the ones generated by this plugin should be retained

Another issue I've encountered is that when running `maven package` consecutively without performing a `maven clean`, I get an error message stating "add-module-info failed: Couldn't copy JAR file."

Original Issue: 
我的项目由于使用了大量未模块化的依赖
于是我使用maven-shade-plugin将依赖全部打包进一个jar内部
不知为何maven-shade-plugin似乎会删掉我自己编写的module-info.class
于是我采用了本项目来重新生成一个module-info.class 
(实际上重新生成才是对的 因为依赖项不写入module-info会在编译期报错 写入后会在运行jar时报错. 全部打包在一起反而可以让jlink认为此项目没有额外依赖项了)

生成时我遇到了一个错误 大意是:"此jar已经是模块化的了"
我解压缩jar包检查发现META-INF\services\9存在module-info.class 但是这个文件是来自于某个依赖jar文件的 最终我排除了这个依赖项才能正常工作
虽然这个问题可能来自maven-shade-plugin 不过按照我的理解 overwriteExistingFiles 为 true 应当能帮助我直接覆盖掉这个文件才对 或者说应当排除所有模块化相关文件 只保留此插件生成的

另外一个问题: 在不进行`maven clean` 的情况下连续使用`maven package` 会出现一个add-module-info failed: Couldn't copy JAR file 错误

``` xml
java version:21
<groupId>org.moditect</groupId>
<artifactId>moditect-maven-plugin</artifactId>
<version>1.2.0.Final</version>

<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.12.1</version>
                <configuration>
                    <source>21</source>
                    <target>21</target>
                    <encoding>UTF-8</encoding>
                    <compilerArgs>
                        <arg>--enable-preview</arg>
                    </compilerArgs>

                </configuration>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-shade-plugin</artifactId>
                <version>3.5.2</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>shade</goal>
                        </goals>
                        <configuration>
                            <createDependencyReducedPom>false</createDependencyReducedPom>
                            <minimizeJar>true</minimizeJar>
                            <transformers>
                                <transformer
                                        implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                    <mainClass>${mainClass}</mainClass>
                                </transformer>
                            </transformers>
                            <filters>
                                <filter>
                                    <artifact>*:*</artifact>
                                    <excludes>
                                        <exclude>module-info.class</exclude>
                                        <exclude>META-INF/*.SF</exclude>
                                        <exclude>META-INF/*.DSA</exclude>
                                        <exclude>META-INF/*.RSA</exclude>
                                    </excludes>
                                </filter>
                            </filters>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.moditect</groupId>
                <artifactId>moditect-maven-plugin</artifactId>
                <version>1.2.0.Final</version>
                <executions>
                    <execution>
                        <id>add-module-infos</id>
                        <phase>package</phase>
                        <goals>
                            <goal>add-module-info</goal>
                        </goals>
                        <configuration>
                            <overwriteExistingFiles>true</overwriteExistingFiles>
                            <module>
                                <moduleInfoSource>
                                    open module testSharedMemory {
                                        requires jdk.httpserver;
                                    }
                                </moduleInfoSource>
                            </module>
                        </configuration>
                    </execution>
                    <execution>
                        <id>create-runtime-image</id>
                        <phase>package</phase>
                        <goals>
                            <goal>create-runtime-image</goal>
                        </goals>
                        <configuration>
                            <modulePath>
                                <path>${project.build.directory}/modules</path>
                            </modulePath>
                            <modules>
                                <module>testSharedMemory</module>
                            </modules>
                            <excludedResources>
                            </excludedResources>
                            <jarInclusionPolicy>NONE</jarInclusionPolicy>
                            <outputDirectory>
                                ${project.build.directory}/jlink-image
                            </outputDirectory>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

**Comment by PhoenixZeng** [comment]
I am currently able to run normally and complete the package and jlink, but I still have doubts about this issue and hope to receive an answer

**Comment by PhoenixZeng** [comment]
Or should other `package` and `JLink` methods be used for my needs like this?

**Comment by aalmiray** [comment]
I think you may be able to instruct the shade plugin to skip entries such as `META-INF/services/*/module-info.class`, you'd have to use a `<filter>` https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html#filters, which I can see in your configuration

**Comment by PhoenixZeng** [comment]
In fact, there is a filter in my plugin and I have excluded it. 
I will still encounter this problem if I change to a path like yours

```
                                <filter>
                                    <artifact>*:*</artifact>
                                    <excludes>
                                        <exclude>module-info.class</exclude> 
                                        <exclude><exclude>META-INF/services/*/module-info.class</exclude>
                                        <exclude>META-INF/*.SF</exclude>
                                        <exclude>META-INF/*.DSA</exclude>
                                        <exclude>META-INF/*.RSA</exclude>
                                    </excludes>
                                </filter>
                            </filters>
```

**Comment by PhoenixZeng** [comment]
Oh, I misread him. His actual path is META-INF/versions/9/module info.class
So the essence of this problem is actually that module info.class will not match inside the folder, and requires a writing style like META INF/versions/*/module info.class

**Comment by PhoenixZeng** [comment]
thanks.  what about the question 2: 

> I've encountered is that when running `maven package` consecutively without performing a `maven clean`, I get an error message stating `add-module-info failed: Couldn't copy JAR file.`

**Comment by aalmiray** [comment]
For the second error that might be a bug in ModiTect but we need more information before we can determina if that's true. 

Please run maven with the `-X` and paste the stacktrace. Be sure to remove any private date before you paste it.

**Comment by PhoenixZeng** [comment]
```
"C:\Program Files\Java\jdk-21\bin\java.exe" -Dmaven.multiModuleProjectDirectory=D:\Java\testSharedMemory -Djansi.passthrough=true "-Dmaven.home=C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\plugins\maven\lib\maven3" "-Dclassworlds.conf=C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\plugins\maven\lib\maven3\bin\m2.conf" "-Dmaven.ext.class.path=C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\plugins\maven\lib\maven-event-listener.jar" "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\lib\idea_rt.jar=12332:C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\bin" -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.stderr.encoding=UTF-8 -classpath "C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\plugins\maven\lib\maven3\boot\plexus-classworlds-2.7.0.jar;C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\plugins\maven\lib\maven3\boot\plexus-classworlds.license" org.codehaus.classworlds.Launcher -Didea.version=2023.2.5 package -X
Apache Maven 3.9.2 (c9616018c7a021c1c39be70fb2843d6f5f9b8a1c)
Maven home: C:\Program Files\JetBrains\IntelliJ IDEA 2023.2.5\plugins\maven\lib\maven3
Java version: 21.0.2, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-21
Default locale: zh_CN, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

// other plugin
[DEBUG] Processing JAR D:\Java\testSharedMemory\target\testSharedMemory-1.jar
[DEBUG] We have a duplicate META-INF/MANIFEST.MF in D:\Java\testSharedMemory\target\testSharedMemory-1.jar
[WARNING] Discovered module-info.class. Shading will break its strong encapsulation.
[DEBUG] Processing JAR C:\Users\Phoenix\.m2\repository\com\github\ShirasawaSama\JavaSharedMemory\0.2.1\JavaSharedMemory-0.2.1.jar
// remove resource
[INFO] Minimized 1770 -> 1380 (77%)
[INFO] Replacing original artifact with shaded artifact.
[INFO] Replacing D:\Java\testSharedMemory\target\testSharedMemory-1.jar with D:\Java\testSharedMemory\target\testSharedMemory-1-shaded.jar
[INFO] 
[INFO] --- moditect:1.2.0.Final:add-module-info (add-module-infos) @ testSharedMemory ---
[DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=13100, ConflictMarker.markTime=8100, ConflictMarker.nodeCount=5, ConflictIdSorter.graphTime=4400, ConflictIdSorter.topsortTime=4900, ConflictIdSorter.conflictIdCount=5, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=51700, ConflictResolver.conflictItemCount=5, DfDependencyCollector.collectTime=5816500, DfDependencyCollector.transformTime=89600}
[DEBUG] org.moditect:moditect-maven-plugin:jar:1.2.0.Final
[DEBUG]    org.eclipse.aether:aether-util:jar:1.1.0:compile
[DEBUG]       org.eclipse.aether:aether-api:jar:1.1.0:compile
[DEBUG]    org.moditect:moditect:jar:1.2.0.Final:compile
[DEBUG]       com.beust:jcommander:jar:1.82:compile
[DEBUG] Created new class realm plugin>org.moditect:moditect-maven-plugin:1.2.0.Final
[DEBUG] Importing foreign packages into class realm plugin>org.moditect:moditect-maven-plugin:1.2.0.Final
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>org.moditect:moditect-maven-plugin:1.2.0.Final
[DEBUG]   Included: org.moditect:moditect-maven-plugin:jar:1.2.0.Final
[DEBUG]   Included: org.moditect:moditect:jar:1.2.0.Final
[DEBUG]   Included: com.beust:jcommander:jar:1.82
[DEBUG] Loading mojo org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info from plugin realm ClassRealm[plugin>org.moditect:moditect-maven-plugin:1.2.0.Final, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@36baf30c]
[DEBUG] Configuring mojo execution 'org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info:add-module-infos' with basic configurator -->
[DEBUG]   (f) artifactId = testSharedMemory
[DEBUG]   (f) buildDirectory = D:\Java\testSharedMemory\target
[DEBUG]   (f) failOnWarning = true
[DEBUG]   (s) moduleInfoSource = open module foo.testSharedMemory {
                                        requires jdk.httpserver;
                                    }
[DEBUG]   (f) module = MainModuleConfiguration [ moduleInfo=null, moduleInfoFile=null, moduleInfoSource=open module foo.testSharedMemory {
                                        requires jdk.httpserver;
                                    }, mainClass=null]
[DEBUG]   (f) outputDirectory = D:\Java\testSharedMemory\target\modules
[DEBUG]   (f) overwriteExistingFiles = true
[DEBUG]   (f) project = MavenProject: foo:testSharedMemory:1 @ D:\Java\testSharedMemory\pom.xml
[DEBUG]   (f) remoteRepos = [jitpack.io (https://www.jitpack.io, default, releases+snapshots), maven - ali (https://maven.aliyun.com/nexus/content/groups/public/, default, releases+snapshots), central (https://repo.maven.apache.org/maven2, default, releases)]
[DEBUG]   (f) repoSession = org.eclipse.aether.DefaultRepositorySystemSession@3ace6346
[DEBUG]   (f) skip = false
[DEBUG]   (f) version = 1
[DEBUG]   (f) workingDirectory = D:\Java\testSharedMemory\target\moditect
[DEBUG] -- end configuration --
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.111 s
[INFO] Finished at: 2024-03-16T03:18:59+08:00
[INFO] ------------------------------------------------------------------------
[WARNING] 
[WARNING] Plugin validation issues were detected in 1 plugin(s)
[WARNING] 
[WARNING]  * org.apache.maven.plugins:maven-resources-plugin:3.3.0
[WARNING] 
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING] 
[ERROR] Failed to execute goal org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info (add-module-infos) on project testSharedMemory: Execution add-module-infos of goal org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info failed: Couldn't copy JAR file: D:\Java\testSharedMemory\target\modules\testSharedMemory-1.jar -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info (add-module-infos) on project testSharedMemory: Execution add-module-infos of goal org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info failed: Couldn't copy JAR file
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:347)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:910)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution add-module-infos of goal org.moditect:moditect-maven-plugin:1.2.0.Final:add-module-info failed: Couldn't copy JAR file
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:910)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.lang.RuntimeException: Couldn't copy JAR file
    at org.moditect.commands.AddModuleInfo.run (AddModuleInfo.java:113)
    at org.moditect.mavenplugin.add.AddModuleInfoMojo.execute (AddModuleInfoMojo.java:238)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:910)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.nio.file.FileAlreadyExistsException: D:\Java\testSharedMemory\target\modules\testSharedMemory-1.jar
    at sun.nio.fs.WindowsException.translateToIOException (WindowsException.java:87)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:103)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:108)
    at sun.nio.fs.WindowsFileSystemProvider.newByteChannel (WindowsFileSystemProvider.java:234)
    at java.nio.file.Files.newByteChannel (Files.java:379)
    at java.nio.file.Files.createFile (Files.java:657)
    at org.moditect.commands.AddModuleInfo.run (AddModuleInfo.java:110)
    at org.moditect.mavenplugin.add.AddModuleInfoMojo.execute (AddModuleInfoMojo.java:238)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:330)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:175)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:76)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:163)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:160)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:910)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[DEBUG] Shutting down adapter factory; available factories [file-lock, rwlock-local, semaphore-local, noop]; available name mappers [discriminating, file-gav, file-hgav, file-static, gav, static]
[DEBUG] Shutting down 'file-lock' factory
[DEBUG] Shutting down 'rwlock-local' factory
[DEBUG] Shutting down 'semaphore-local' factory
[DEBUG] Shutting down 'noop' factory

```

**Comment by aalmiray** [comment]
Shipped in `1.2.1.Final`.
